client/my-sites/sidebar/sidebar.jsx

```diff
@@ -83,7 +82,7 @@ module.exports = React.createClass( {
 		}
 
 		return paths.some( function( path ) {
-			return startsWith( this.props.path, path );
+			return ( new RegExp( '^' + path + '(\\/|$)' ) ).test( this.props.path );
```

Comment:
>Would we ever have a path that contains reserved regex characters?

That's a good point. I may try to solve this with an alternative approach avoiding the regular expression.